src/ClientIp.php

```diff
@@ -107,33 +107,70 @@ public function process(ServerRequestInterface $request, DelegateInterface $dele
      */
     private function getIp(ServerRequestInterface $request)
     {
-        if ($this->remote) {
-            $ip = file_get_contents('http://ipecho.net/plain');
+        if ($ip = $this->getRemoteIp()) {
+            return $ip;
+        }
 
-            if (self::isValid($ip)) {
+        if ($ip = $this->getProxyIp($request)) {
+            if (!in_array($ip, $this->proxyIps)) {
                 return $ip;
             }
         }
 
-        $ip = null;
-        $server = $request->getServerParams();
-
-        if (!empty($server['REMOTE_ADDR']) && self::isValid($server['REMOTE_ADDR'])) {
-            $ip = $server['REMOTE_ADDR'];
+        if ($ip = $this->getLocalIp($request)) {
+            if (!in_array($ip, $this->proxyIps)) {
+                return $ip;
+            }
```

Comment:
I think the functionality does not remain the same. The logic should be:
* Get the ip using the default way (`REMOTE_ADDR`). Let's call it `$localIp`
* If proxies hasn't been configured returns `$localIp`
* If there's an array of allowed proxy ips, check if `$localIp` is in this array. If it's not, returns `$localIp`.
* Get the new ip using the proxy headers and return it if it's found, or `$localIp` if doesn't.

In this refactoring `$this->proxyIps` is used to check the ip returned by `$this->getProxyIp()`, it should used to check the ip returned by `$this->getLocalIp()`